khmer/_khmermodule.cc

```diff
@@ -4201,6 +4202,190 @@ static void khmer_subset_dealloc(PyObject* self)
     PyObject_Del((PyObject *) obj);
 }
 
+
+/***********************************************************************/
+
+//
+// KHLLCounter object
+//
+
+typedef struct {
+    PyObject_HEAD
+    khmer::HLLCounter * hllcounter;
+} khmer_KHLLCounterObject;
+
+static void khmer_hllcounter_dealloc(PyObject *);
+static PyObject * khmer_hllcounter_getattr(PyObject * obj, char * name);
+
+static PyTypeObject khmer_KHLLCounterType = {
+    PyObject_HEAD_INIT(NULL)
+    0,
+    "KHLLCounter", sizeof(khmer_KHLLCounterObject),
+    0,
+    khmer_hllcounter_dealloc,	/*tp_dealloc*/
+    0,				/*tp_print*/
+    khmer_hllcounter_getattr,	/*tp_getattr*/
+    0,				/*tp_setattr*/
+    0,				/*tp_compare*/
+    0,				/*tp_repr*/
+    0,				/*tp_as_number*/
+    0,				/*tp_as_sequence*/
+    0,				/*tp_as_mapping*/
+    0,				/*tp_hash */
+    0,				/*tp_call*/
+    0,				/*tp_str*/
+    0,				/*tp_getattro*/
+    0,				/*tp_setattro*/
+    0,				/*tp_as_buffer*/
+    Py_TPFLAGS_DEFAULT,		/*tp_flags*/
+    "HyperLogLog counter object",           /* tp_doc */
+};
+
+#define is_hllcounter_obj(v)  ((v)->ob_type == &khmer_KHLLCounterType)
+
+//
+// new_hll_counter
+//
+
+static PyObject* _new_hll_counter(PyObject * self, PyObject * args)
+{
+    double error_rate = 0;
+    WordLength ksize = 0;
+
+    if (!PyArg_ParseTuple(args, "db", &error_rate, &ksize)) {
+        return NULL;
+    }
+
+    if ((error_rate < 0) || (error_rate > 1.0)) {
+        PyErr_SetString(PyExc_ValueError, "Error rate should be between 0.0 and 1.0");
+        return NULL;
+    }
+
+    khmer_KHLLCounterObject * khllcounter_obj = (khmer_KHLLCounterObject *) \
+            PyObject_New(khmer_KHLLCounterObject, &khmer_KHLLCounterType);
+
+    if (khllcounter_obj == NULL) {
+        return NULL;
+    }
+
+    try {
+        khllcounter_obj->hllcounter = new khmer::HLLCounter(error_rate, ksize);
+    } catch (khmer_exception &e) {
+        PyErr_SetString(PyExc_ValueError, e.what());
+        return NULL;
+    }
+
+    return (PyObject *) khllcounter_obj;
+}
+
+//
+// khmer_hllcounter_dealloc -- clean up a hllcounter object.
+//
+
+static void khmer_hllcounter_dealloc(PyObject* self)
+{
+    khmer_KHLLCounterObject * obj = (khmer_KHLLCounterObject *) self;
+    delete obj->hllcounter;
+    obj->hllcounter = NULL;
+
+    PyObject_Del((PyObject *) obj);
+}
+
+static
+PyObject *
+hllcounter_add( PyObject * self, PyObject * args )
```

Comment:
This is useful for testing and it is an easy way to prototype further functionality, so I think it can stay.